packages/dds/tree/src/test/simple-tree/benchmarkUtilities.ts

```diff
@@ -0,0 +1,151 @@
+	return depth === 1
+		? new DeepTreeNode({ node: undefined, value: leafValue })
+		: new DeepTreeNode({ node: makeDeepTree(depth - 1, leafValue), value: undefined });
+}
+
+/**
+ * generate a deep tree with a single leaf value
+ */
+export function generateDeepSimpleTree(depth: number, leafValue: number): DeepTreeNode {
+	if (depth < 1) {
+		throw new Error("Depth must be greater than 0");
+	}
+
+	return hydrate(DeepTreeNode, makeDeepTree(depth, leafValue));
+}
+
+/**
+ * read the deep tree and return the depth and value of the leaf node
+ */
+export function readDeepSimpleTree(tree: DeepTreeNode): {
+	depth: number;
+	value: number;
+} {
+	let currentNode = tree;
+	let depth = 1;
+
+	while (currentNode.node !== undefined) {
+		depth += 1;
+		currentNode = currentNode.node;
+	}
+
+	const value = currentNode.value ?? assert.fail("missing leaf value");
+	return { depth, value };
+}
+
+/**
+ * Update the deep tree with a new value
+ */
+export function writeDeepSimpleTree(tree: DeepTreeNode, newValue: number): void {
+	let currentNode = tree;
+
+	while (currentNode.node !== undefined) {
+		currentNode = currentNode.node;
+	}
+
+	currentNode.value = newValue;
+}
+
+/**
+ * Make a wide tree with a single layer
+ */
+function makeWideTree(width: number, leafValue: number): WideTreeNode {
+	const values = [];
+	for (let i = 0; i < width; i++) {
+		values.push(leafValue);
+	}
+	return new WideTreeNode({ values });
+}
+
+/**
+ * generate a wide tree with a single layer
+ */
+export function generateWideSimpleTree(width: number, leafValue: number): WideTreeNode {
+	return hydrate(WideTreeNode, makeWideTree(width, leafValue));
+}
+
+/**
+ * read the whole wide tree and return the width and the sum of all value in the leaf node
+ */
+export function readWideSimpleTree(tree: WideTreeNode): {
+	nodesCount: number;
+	sum: number;
+} {
+	const nodesCount = tree.values.length;
+	let sum = 0;
+	for (const value of tree.values) {
+		sum += value;
+	}
+	return { nodesCount, sum };
+}
+
+/**
+ * write the whole wide tree with a new value
+ */
+export function writeWideSimpleTree(tree: WideTreeNode, newValue: number): void {
+	const nodesCount = tree.values.length;
+	for (let i = 0; i < nodesCount; i++) {
+		tree.values.removeAt(i);
+		tree.values.insertAt(i, newValue);
+	}
+}
+
+/**
+ * read the last value of the wide tree
+ */
+export function readWideSimpleTreeValue(tree: WideTreeNode, index: number): number {
+	return tree.values[index];
+}
+
+/**
+ * Update the input index value for the wide tree.
+ */
+export function writeWideSimpleTreeNewValue(
+	tree: WideTreeNode,
+	newValue: number,
+	index: number,
+): void {
+	tree.values.insertAt(index, newValue);
+	tree.values.removeAt(index + 1);
+}
+
+/**
+ * Move an item in an array node to a new position.
+ */
+export function moveWideSimpleTreeValue(
+	tree: WideTreeNode,
+	fromIndex: number,
+	toIndex: number,
+): void {
+	tree.values.moveToIndex(toIndex, fromIndex);
+}
```

Comment:
I don't think this functional adds any value (especially after simplifying the WideTreeNode schema) since it just wraps a method on the passed in object. The caller could just as easily do tree.moveToIndex(dst, src), which is the public APi we want to test.

Reply:
Another reason to remove helpers like this is the it avoids needing to document the parameters: if the test called moveToIndex directly, then the author and readers of the perf test would have intelisense for the paramaters with the full documentation, which they don't get with this method.

Also, since its a perf test, adding an extra wrapper adds a little overhead when its easy to write the perf test against the actual API thats preferred to avoid such overhead.

Reply:
I see. Remove it then